docs/decisions/2024-07-10_include_file_limits.md

```diff
@@ -0,0 +1,17 @@
+# 2024-03-26 Appropriate limit for include files per project
+
+- status: in review (will change to accepted after PR is approved)
+- date: 2024-07-11
+- authors: Bynn Lee
+
+## Context and Problem Statement
+
+The introduction of the `include` capability to project configuration has increased Evergreen's use of git clones since we ask GitHub for each individual included file separately. We needed a way to stop users from infinitely scaling the number of includes until we hit our GitHub API limit. 
+
+Include files take up a significant percentage of all of Evergreen's GitHub operations. After switching to GitHub apps, our new hourly limit is 15,000 requests to GitHub per hour. On average, we only use around 4-8000 of our limit per hour, depending on how active users are. Of those requests, around 60% can be contributed to retrieving include files. For example, on an hour we used 7000 of our limit, we would also have around 4000 include file retrieval requests logged. 
+
+The two biggest loads are the the mongo repo projects that have around 25 includes on average, and the mms repo projects which currently have 32 include files. However, the mongo projects have not added extra include files in the past year whereas the mms projects have added almost 10 new include files.
+
+## Decision Outcome
+
+I propose a limit of 40 includes on all projects. Assuming our GitHub usage scales consistently with what we have been seeing, we should be able to handle a 100% increase on the number of include files. So in theory, a limit of 50 should not exceed our limits. However, there are intermittent cases (about once a month) where we do end up using 10,000 of our API limit for the hour and setting the limit too high could use up our limit during those times. 
```

Comment:
I'm skeptical that we can tolerate a doubling of the number of include files if there are hours (as mentioned above) where we use 7000 requests in an hour for includes, and this only represents around 60% of total requests?

If anything, I'd consider pushing the limit even lower to e.g. 35. Including 35 files is quite a lot and I'd think cloud and/or server should be able to keep their includes at this number. If there is a reason they need to exceed that, it might be worthwhile to try to chat with them about it to better understand their intentions. CC @ablack12

Reply:
When there are 7,000 total requests, there are around 4,000 include requests wthich is why (in theory) we could potentially handle a larger number of them but I do think 35 is a lot more reasonable and leaves room for company growth with more engineers as @Kimchelly was saying. 

I'm not sure who we should talk to about this but I think 35 is a good limit if we could properly communicate it to the teams writing the configurations so that the limit won't come out of nowhere.